core/src/main/scala/com/databricks/labs/remorph/intermediate/expressions.scala

```diff
@@ -161,7 +161,25 @@ case class Cast(
     type_str: String = "",
     returnNullOnError: Boolean = false,
     timeZoneId: Option[String] = None)
-    extends Unary(expr)
+    extends Unary(expr) {
+
+  override def makeCopy(newArgs: Array[AnyRef]): Expression = {
```

Comment:
Existing tests crashing? How's that not caught by CI?

Reply:
A very similar issue is reproduced in `core/src/test/scala/com/databricks/labs/remorph/intermediate/TreeTest.scala` in branch `fix-makecopy`.
Strangely the fixes in this branch don't fix it.
I suspect the issue might come from using `Array` instead of `Seq`, and unlike the Scala compiler, `makeCopy` doesn't implement the required conversion

Reply:
@vil1 
Further investigation shows that the issue originated from:

```
private def dealiasUnary(unary: Unary): Expression = {
    val child = dealiasExpression(unary.child)
    unary.makeCopy(Array(child))
  } 

```
All 3 problematic classes (`Alias`, `Cast` and `WithinGroup`) inherit from Unary but actually have more than 1 field